src/login/logind-dbus.c

```diff
@@ -2038,6 +2044,94 @@ static usec_t nologin_timeout_usec(usec_t elapse) {
         return LESS_BY(elapse, 5 * USEC_PER_MINUTE);
 }
 
+int manager_load_scheduled_shutdown(Manager *m) {
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *usec = NULL,
+               *warn_wall = NULL,
+               *mode = NULL,
+               *wall_message = NULL,
+               *uid = NULL,
+               *tty = NULL;
+        _cleanup_free_ char *unescaped = NULL;
+        int r;
+
+        assert(m);
+
+        r = parse_env_file(f, SHUTDOWN_SCHEDULE_FILE,
+                        "USEC", &usec,
+                        "WARN_WALL", &warn_wall,
+                        "MODE", &mode,
+                        "WALL_MESSAGE", &wall_message,
+                        "UID", &uid,
+                        "TTY", &tty);
+        if (r == -ENOENT)
+                return 0;
+        if (r < 0) {
+                log_debug_errno(r, "Failed to parse " SHUTDOWN_SCHEDULE_FILE ": %m");
+                goto fail;
+        }
+
+        r = deserialize_usec(strnull(usec), &m->scheduled_shutdown_timeout);
+        if (r < 0)
+                goto fail;
+
+        HandleAction handle = handle_action_from_string(mode);
+        if (handle == _HANDLE_ACTION_INVALID) {
+                log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse scheduled shutdown type '%s'", mode);
+                r = -EINVAL;
+                goto fail;
+        }
+        m->scheduled_shutdown_type = manager_item_for_handle(handle);
+
+        if (wall_message) {
+                r = cunescape(wall_message, 0, &unescaped);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to parse wall message");
+                        goto fail;
+                }
+                if (free_and_strdup_warn(&m->wall_message, empty_to_null(unescaped)) < 0)
+                        goto fail;
+        }
+
+        if (uid) {
+                r = parse_uid(uid, &m->scheduled_shutdown_uid);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to parse wall uid");
+                        goto fail;
+                }
+        }
+
+        if (free_and_strdup_warn(&m->scheduled_shutdown_tty, empty_to_null(tty)) < 0)
+                goto fail;
+
+        /* do this last as we'd have to reset on fail and we do not want to repeat any built in default */
+        if (warn_wall) {
+                r = parse_boolean(warn_wall);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to parse enabling wall messages");
+                        goto fail;
+                }
+                m->enable_wall_messages = r;
+        }
+
+        r = manager_setup_shutdown_timers(m);
+        if (r < 0)
+                goto fail;
+
+        r = manager_setup_wall_message_timer(m);
+
+        return r;
+
+fail:
+        m->scheduled_shutdown_timeout = 0;
```

Comment:
USEC_INFINITY?

Reply:
those scheduled_* settings are not separately initialized (see https://github.com/systemd/systemd/blob/c972880640ee19e89ce9265d8eae1b3aae190332/src/login/logind.c#L44), so the uninitialized values are zero.

Reply:
`reset_scheduled_shutdown()` could probably do that initialization. Then we call it on startup and in the fail case